book/data_viz/plotting_data.nim

```diff
@@ -7,18 +7,260 @@ nbUseNimibook
 nbText: """
 # Plotting data using [ggplotnim](https://github.com/Vindaar/ggplotnim)
 
-We'll plot some pretty plots!
+In this tutorial we will introduce `ggplotnim`, a Nim plotting library heavily inspired
+by the great R library [ggplot2](https://ggplot2.tidyverse.org).
 
-## Get df
+This will be kept rather brief, but we will discuss the philosophy of the syntax, look
+at a reasonably complex plotting example that we deconstruct and finish of by (hopefully)
+coming to the conclusion that the ggplot-like syntax is rather elegant.
 
-Get some DF in here.
+For this tutorial you should have read the data wrangling introduction to `Datamancer` or
+know about data frames and have seen the `Datamancer` formula macro `f{}`.
 
-## plot something
+## On philosophy and graphics
 
-And plot something!
+Similar to most areas of life touched by more than a few people who seemingly all have
+their own ideas about the right way to do things, plotting libraries come in different
+shapes and forms. In terms of their output formats, choice of colors and style and
+of most importance for us here: in terms of their API / the programming syntax used.
+
+Most plotting libraries fall into a category that are either focused on object orientation
+(your commands return some objects for you to modify to your needs) or a generally imperative
+style (call this function `plotFoo` for plot style A, that function `plotBar` for style B,
+etc.) and often some combination of these two.
+
+`ggplot2` and as a result `ggplotnim` follow a declarative style that builds up a plot
+from a single command by combining multiple different layers as a chain of commands.
+
+This is because `ggplot2` is an implementation of the so called "grammar of graphics".
+It's the NixOS of plotting libraries. Tell it what you want and it gets it done for you,
+as long as you speak its language.
+
+## A motivating example
+
+Let's now consider a somewhat complicated plotting example. Using that we will look at why
+it is called a *grammar* of graphics.
+"""
+block:
```

Comment:
to test the preview thing with a simple commit you can try replacing this with `nbCodeInBlock` which is available now in latest nimib 😄

Reply:
Ah, right. Good idea! :+1: